src/functionalTest/java/uk/gov/hmcts/dts/fact/admin/list/AdminLocalAuthoritiesEndpointTest.java

```diff
@@ -0,0 +1,122 @@
+package uk.gov.hmcts.dts.fact.admin.list;
+
+import io.restassured.response.Response;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.dts.fact.model.admin.LocalAuthority;
+import uk.gov.hmcts.dts.fact.util.AdminFunctionalTestBase;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpHeaders.*;
+import static org.springframework.http.HttpStatus.*;
+import static uk.gov.hmcts.dts.fact.util.TestUtil.*;
+
+@ExtendWith(SpringExtension.class)
+public class AdminLocalAuthoritiesEndpointTest extends AdminFunctionalTestBase {
+
+    private static final String LOCAL_AUTHORITIES_ENDPOINT = "/admin/localauthorities";
+    private static final String GET_ALL_LOCAL_AUTHORITIES_ENDPOINT = LOCAL_AUTHORITIES_ENDPOINT + "/all";
+    private static final String BIRMINGHAM_CITY_COUNCIL = "Birmingham City Council";
+    private static final String NON_EXISTENT_LOCAL_AUTHORITY = "Briminghm Council";
+
+    @Test
+    public void shouldGetAllLocalAuthorities() {
+        final Response response = doGetRequest(GET_ALL_LOCAL_AUTHORITIES_ENDPOINT, Map.of(AUTHORIZATION, BEARER + authenticatedToken));
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        final List<LocalAuthority> localAuthorities = response.body().jsonPath().getList(".", LocalAuthority.class);
+        assertThat(localAuthorities).hasSizeGreaterThan(1);
+    }
+
+    @Test
+    public void shouldRequireATokenWhenGettingAllLocalAuthorities() {
+        final Response response = doGetRequest(GET_ALL_LOCAL_AUTHORITIES_ENDPOINT);
+        assertThat(response.statusCode()).isEqualTo(UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void shouldBeForbiddenForGettingLocalAuthorities() {
+        final Response response = doGetRequest(GET_ALL_LOCAL_AUTHORITIES_ENDPOINT, Map.of(AUTHORIZATION, BEARER + forbiddenToken));
+        assertThat(response.statusCode()).isEqualTo(FORBIDDEN.value());
+    }
+
+    @Test
+    public void shouldUpdateExistingLocalAuthority() {
```

Comment:
Just checking...in this test we are not actually updating any name? just updating to the same one?

Reply:
Yes. This is deliberate. Whilst ideally we would check we can set it to a different value, there are 2 issues with trying to do this in a functional test:

- It has to be a valid local authority otherwise the mapit validation will fail. I can't find another valid local authority to use that doesn't exist in the DB.
- We have a duplicate check. We'd need to delete an entry (there is no endpoint to do that) or keep changing to a different value to avoid the duplicate validation causing it to fail.